pkg/controller/controller_utils.go

```diff
@@ -1032,6 +1033,52 @@ func UpdateNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node,
 	return nil
 }
 
+// AddOrUpdateAnnotationsOnNode add annotations to the node. If annotations was added into node, it'll issue API calls
+// to update nodes; otherwise, no API calls. Return error if any.
+func AddOrUpdateAnnotationsOnNode(c clientset.Interface, nodeName string, annotations map[string]string) error {
+	if annotations == nil {
+		return nil
+	}
+	firstTry := true
+	return clientretry.RetryOnConflict(UpdateNodeBackoff, func() error {
+		var err error
+		var oldNode *v1.Node
+		if firstTry {
+			oldNode, err = c.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{ResourceVersion: "0"})
+			firstTry = false
+		} else {
+			oldNode, err = c.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+		}
+
+		if err != nil {
+			return err
+		}
+
+		var newNode *v1.Node
+		oldNodeCopy := oldNode
+		for key, value := range annotations {
+			curNewNode := annotationutils.AddOrUpdateAnnotation(oldNodeCopy, key, value)
+			newNode = curNewNode
+			oldNodeCopy = curNewNode
+		}
+		return UpdateNodeAnnotations(c, nodeName, oldNode, newNode)
+	})
+}
+
+// UpdateNodeAnnotations is for updating the node annotations from oldNode to the newNode
+// using the nodes Update() method
+func UpdateNodeAnnotations(c clientset.Interface, nodeName string, oldNode *v1.Node, newNode *v1.Node) error {
+	newNodeClone := oldNode.DeepCopy()
+	newNodeClone.ObjectMeta.Annotations = newNode.ObjectMeta.Annotations
+
+	_, err := c.CoreV1().Nodes().Update(newNodeClone)
+	if err != nil {
```

Comment:
Hi @ggaurav10  - I was keeping with the conventions of the other utils here: `NotFound` errors are retried, but the caller suppresses the error to avoid re-queuing of the machine. This is less limiting for future consumers that may want to handle the `NotFound` error differently when using the annotation util. Is that acceptable to you?

Reply:
sure, sounds good. Just wanted to make sure if a custom error like the one sent in this function would also result in the desired behaviour in case of `NotFound` error.